[RA2 Tech] Change containeri[s|z]ed to cloud native where appropriate (#1652)
cntt-n/CNTT: doc/ref_arch/kubernetes/chapters/chapter05.md

- Integrate run-time processes to Security Information and Event Monitoring (SIEM)
- Use container-aware run-time defense tools

> - Ensure all running container applications are from secure and verified images

Reviewer: ```suggestion
- Ensure all running cloud native applications are from secure and verified images
```

> - Cloud native application should not run with root privileges

Reviewer: ```suggestion
- Cloud native applications should not run with root privileges
```

- Ensure sensitive workloads are properly segmented by namespaces or cluster to mitigate the scope of compromise.

##  5.12  Secrets Management
The principle of least privilege must be applied to secret management in Kubernetes;

> - Ensure the cloud native application code can read only the secrets that it needs

Reviewer: ```suggestion
- Ensure that the cloud native applications can only read the secrets that these applications need
```

- Containers: isolation by SW based mechanisms available in OS, Docker and Kubernetes. A container workload is just a set of Linux processes. It is _possible_ to configure SW based _additional isolation_ for container workloads, for example by kernel namespaces.

> Thus the primary isolation mechanism in Kubernetes environment should be VM or physical machine based isolation. This means: multiple container applications should not be deployed together in the same Kubernetes cluster - unless those have been planned and verified to co-exist. In such cases, the default is to allocate one namespace per Cloud Native Network Function (CNF).

Reviewer: ```suggestion
The primary isolation mechanism in Kubernetes environment should be VM or physical machine based. This implies that multiple cloud native applications should not be deployed together in the same Kubernetes cluster - unless these applications have been planned and verified to co-exist. Thus, the default is to allocate one namespace per Cloud Native Network Function (CNF).
```

### Container Isolation in Kubernetes Cluster
#### Namespaces  

> Kubernetes namespaces should be used to provide resource isolation within a Kubernetes cluster. They should not be used to isolate different steps in the deployment process like Development, Production, or Testing. The most reliable separation is achieved by deploying sensitive workloads into dedicated clusters.

Reviewer: Are "Development, Production, or Testing" steps in deployment process or they are different stages in the application development process.

Author: Not sure - note, I didn't actually make any changes here so would prefer to handle this in a separate issue/PR please.
